libvcx/src/aries/handlers/connection/invitee/state_machine.rs

```diff
@@ -193,74 +187,162 @@ impl SmConnectionInvitee {
+            InviteeState::Invited(state) => {
+                let new_agent_info = agent_info.create_agent()?;
+                let request = Request::create()
+                    .set_label(source_id.to_string())
+                    .set_did(new_agent_info.pw_did.to_string())
+                    .set_service_endpoint(new_agent_info.agency_endpoint()?)
+                    .set_keys(new_agent_info.recipient_keys(), new_agent_info.routing_keys()?);
+
+                trace!("invitation {:?}", state.invitation);
+                let ddo = DidDoc::from(state.invitation.clone());
+                ddo.send_message(&request.to_a2a_message(), &new_agent_info.pw_vk)?;
+                let new_state = InviteeState::Requested((state, request).into());
+                (new_state, new_agent_info)
+            },
+            _ => {
+                (state.clone(), agent_info)
+            }
+        };
+        Ok(SmConnectionInvitee { source_id, agent_info: new_agent_info, state: new_state })
+    }
+
+    pub fn transition_receive_invitation(self, invitation: Invitation) -> VcxResult<SmConnectionInvitee>  {
+        let SmConnectionInvitee { source_id, agent_info, state } = self;
+        let new_state = match state {
             InviteeState::Null(state) => {
-                match message {
-                    DidExchangeMessages::InvitationReceived(invitation) => {
-                        InviteeState::Invited((state, invitation).into())
-                    }
-                    _ => {
-                        InviteeState::Null(state)
-                    }
-                }
+                InviteeState::Invited((state, invitation).into())
             }
-            InviteeState::Invited(state) => {
-                match message {
-                    DidExchangeMessages::Connect() => {
-                        agent_info = agent_info.create_agent()?;
-                        let request = Request::create()
-                            .set_label(source_id.to_string())
-                            .set_did(agent_info.pw_did.to_string())
-                            .set_service_endpoint(agent_info.agency_endpoint()?)
-                            .set_keys(agent_info.recipient_keys(), agent_info.routing_keys()?);
-
-                        trace!("invitation {:?}", state.invitation);
-                        let ddo = DidDoc::from(state.invitation.clone());
-                        ddo.send_message(&request.to_a2a_message(), &agent_info.pw_vk)?;
-                        InviteeState::Requested((state, request).into())
-                    }
-                    DidExchangeMessages::ProblemReportReceived(problem_report) => {
-                        InviteeState::Null((state, problem_report).into())
-                    }
-                    _ => {
-                        InviteeState::Invited(state)
-                    }
-                }
+            _ => {
+                state.clone()
```

Comment:
According to the [connection RFC](https://github.com/hyperledger/aries-rfcs/tree/master/features/0160-connection-protocol), receiving an invitation in requested state should start a new relationship, and probably return error in responded and complete. Of course, there are more differences like this (some addressed in my backchannel PR, some not covered by the harness tests ATM) we should probably leave to another PR, though.